src/identityApiClient.ts

```diff
@@ -99,55 +114,76 @@ export default function IdentityAPIClient(
-                        'Successfully sent forwarding stats to mParticle Servers';
-                    break;
-                default:
-                    // 400 has an error message, but 403 doesn't
-                    if (aliasResponseBody?.message) {
-                        errorMessage = aliasResponseBody.message;
+                case HTTP_OK:
+
+                // 400 error will has a body and will go through the happy path to report the error
+                case HTTP_BAD_REQUEST:
+
+                    // FetchUploader returns the response as a JSON object that we have to await
+                    if (response.json) {
+                        // HTTP responses of 202, 200, and 403 do not have a response.
+                        // response.json will always exist on a fetch, but can only be
+                        // await-ed when the response is not empty, otherwise it will
+                        // throw an error.
+                        try {
+                            aliasResponseBody = await response.json();
+                        } catch (e) {
+                            verbose('The request has no response body');
+                        }
+                    } else {
+                        // https://go.mparticle.com/work/SQDSDKS-6568
+                        // XHRUploader returns the response as a string that we need to parse
+                        const xhrResponse = (response as unknown) as XMLHttpRequest;
+
+                        aliasResponseBody = xhrResponse.responseText
+                            ? JSON.parse(xhrResponse.responseText)
+                            : '';
+
+                        // https://go.mparticle.com/work/SQDSDKS-6670
+                        message =
+                            'Successfully sent forwarding stats to mParticle Servers';
                     }
-                    message =
-                        'Issue with sending Alias Request to mParticle Servers, received HTTP Code of ' +
-                        response.status;
+
+
+                    if (response.status === HTTP_BAD_REQUEST) {
+                        // 400 has an error message, but 403 doesn't
+                        const errorResponse: IAliasErrorResponse = aliasResponseBody as unknown as IAliasErrorResponse;
+                        if (errorResponse?.message) {
+                            errorMessage = errorResponse.message;
+                        }
+                        message =
+                            'Issue with sending Alias Request to mParticle Servers, received HTTP Code of ' +
+                            response.status;
+                        
+                        if (errorResponse?.code) {
+                            message += ' - ' + errorResponse.code;
+                        }
+                    }
+
+                    break;
+                    
+                // 401 and 403 have no bodies and should be rejected outright
```

Comment:
Note: Rephrase to say 'All other error codes will hit this path'

Reply:
Please update this